example/game/src/main/java/no/nordicsemi/android/ble/example/game/server/viewmodel/ServerViewModel.kt

```diff
@@ -0,0 +1,234 @@
+            .launchIn(viewModelScope)
+        _selectedAnswer.value = null
+        _correctAnswerId.value = null
+        _state.value = Round(question)
+    }
+
+    private fun startServer() {
+        advertiser.startAdvertising()
+        serverManager.setServerObserver(object : ServerObserver {
+
+            override fun onServerReady() {
+                Log.w("Start Server", "Server is Ready!!")
+            }
+
+            override fun onDeviceConnectedToServer(device: BluetoothDevice) {
+                ServerConnection(context, viewModelScope, device)
+                    .apply {
+                        stateAsFlow()
+                            .onEach { connectionState ->
+                                val currentState = _state.value
+                                when (connectionState) {
+                                    ConnectionState.Ready -> {
+                                        clients.value += this
+                                        _state.value = WaitingForPlayers(clients.value.size)
+                                    }
+                                    is ConnectionState.Disconnected -> {
+                                        clients.value -= this
+
+                                        when (currentState) {
+                                            is WaitingForPlayers -> {
+                                                _state.value = WaitingForPlayers(clients.value.size)
+                                            }
+                                            else -> {
+                                                // TODO handle disconnection during game
+                                            }
+                                        }
+                                    }
+                                    else -> {}
+                                }
+                            }
+                            .launchIn(viewModelScope)
+                    }
+                    .apply {
+                        playersName
+                            .onEach { savePlayersName(it) }
+                            .launchIn(viewModelScope)
+                    }
+                    .apply {
+                        result
+                            .onEach { saveScore(it) }
+                            .launchIn(viewModelScope)
+                    }
+                    .apply {
+                        useServer(serverManager)
+
+                        val exceptionHandler = CoroutineExceptionHandler { _, throwable ->
+                            Log.e(TAG, "Error", throwable)
+                        }
+                        viewModelScope.launch(Dispatchers.IO + exceptionHandler) {
+                            connect()
+                        }
+                    }
+
+            }
+
+            override fun onDeviceDisconnectedFromServer(device: BluetoothDevice) {
+                Log.w("Device Disconnected", "Device Disconnected From the Server!")
```

Comment:
This is not handled. I think server could at least remove the `ServerConnection` for this device if game is in progress?